driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -2208,16 +2208,35 @@ public void ensurePoolsSizing() {
         }
 
         public PreparedStatement addPrepared(PreparedStatement stmt) {
```

Comment:
Do we still need to change this method?
It's not very important to handle race conditions here; even if we store an instance that has the wrong `resultSetMetadata`, the server will inform us in the first response and we'll mutate the instance then.